### PR TITLE
Add UWP test runner

### PR DESCRIPTION
**Pull request by nirinchev**
## Description

Part of #1081

### CODE REVIEW

File: Realm/Realm/Linq/RealmResultsVisitor.cs [hunk 1]
```diff
@@ -449,6 +449,39 @@ internal override Expression VisitMethodCall(MethodCallExpression m)
             throw new NotSupportedException($"The method '{m.Method.Name}' is not supported");
         }
 
+        // Compares two methods for equality. .NET Native's == doesn't return expected results.
+        private static bool AreMethodsSame(MethodInfo first, MethodInfo second)
```
Comment (nirinchev) [inline Realm/Realm/Linq/RealmResultsVisitor.cs]: 👀

File: Realm/Realm/Realm.cs [hunk 2]
```diff
@@ -368,7 +368,7 @@ public override int GetHashCode()
         {
             ThrowIfDisposed();
 
-            return (int)SharedRealmHandle.DangerousGetHandle();
+            return (int)((long)SharedRealmHandle.DangerousGetHandle() % int.MaxValue);
```
Comment (nirinchev) [inline Realm/Realm/Realm.cs]: 👀

File: Tests/Tests.Shared/DateTimeTests.cs [hunk 3]
```diff
@@ -28,11 +28,8 @@ namespace Tests.Database
     public class DateTimeTests : RealmInstanceTest
     {
         [Test]
-        public void SetAndGetPropertyTest(
-            [Values(0, 11, 23)] int hour,
-            [Values(0, 6, 30, 59)] int mins,
-            [Values(0, 6, 30, 59)] int secs,
-            [Values(0, 1, 999)] int ms)
+        [TestCaseSource(nameof(SetAndGetPropertyTestCases))]
```
Comment (nirinchev) [inline Tests/Tests.Shared/DateTimeTests.cs]: .NET Native fails to correctly pass attributes with arrays

File: Realm/Realm/Handles/ObjectHandle.cs [hunk 4]
```diff
@@ -519,20 +519,34 @@ public unsafe void SetByteArray(IntPtr propertyIndex, byte[] value)
 
         public byte[] GetByteArray(IntPtr propertyIndex)
         {
+            return GetByteArrayBuffer(propertyIndex, 0);
+        }
+
+        private unsafe byte[] GetByteArrayBuffer(IntPtr propertyIndex, int size)
```
Comment (nirinchev) [inline Realm/Realm/Handles/ObjectHandle.cs]: 👀

File: Realm/Realm/Native/NativeCommon.cs [hunk 5]
```diff
@@ -69,10 +69,17 @@ public static unsafe void Initialize()
...
-                    var path = Path.Combine(assemblyLocation, "lib", "win32", architecture) + Path.PathSeparator + Environment.GetEnvironmentVariable("PATH");
-                    Environment.SetEnvironmentVariable("PATH", path);
+                    try
+                    {
+                        var assemblyLocation = Path.GetDirectoryName((string)assemblyLocationPI.GetValue(typeof(NativeCommon).GetTypeInfo().Assembly));
+                        var architecture = InteropConfig.Is64BitProcess ? "x64" : "x86";
+                        var path = Path.Combine(assemblyLocation, "lib", "win32", architecture) + Path.PathSeparator + Environment.GetEnvironmentVariable("PATH");
+                        Environment.SetEnvironmentVariable("PATH", path);
+                    }
+                    catch (PlatformNotSupportedException)
```
Comment (nirinchev) [inline Realm/Realm/Native/NativeCommon.cs]: I was unable to figure out a way to determine whether we're UWP or regular Windows Desktop app :/ The hack with checking for a Windows Runtime type doesn't work, as you can reference WindowsRuntime in a regular app. Suggestions are welcome.

Comment (fealebenpae) [inline Realm/Realm/Native/NativeCommon.cs]: Well, if it works with the `try/catch` let's leave it as it is.

File: Realm/Realm/Realm.cs (hunk 2 above)
Comment (kristiandupont) [inline Realm/Realm/Realm.cs]: What's going on here?

Comment (nirinchev) [inline Realm/Realm/Realm.cs]: `DangerousGetHandle()` returns `IntPtr` which is long on 64 bit devices. I'm not sure how it has worked until now, but on UWP it consistently throws out of range exception. There are different strategies for wrapping it in an int - I chose one that seemed fine, but I'm open to suggestions if anyone has a better idea :)

Comment (kristiandupont) [inline Realm/Realm/Realm.cs]: Ah, I'm seeing now that it's in `GetHashCode`. No worries then

File: Realm/Realm.Sync/Realm.Sync.csproj [hunk 6]
```diff
@@ -80,4 +80,5 @@
   </ItemGroup>
   <Import Project="$(MSBuildExtensionsPath32)\Microsoft\Portable\$(TargetFrameworkVersion)\Microsoft.Portable.CSharp.targets" />
   <Import Project="..\..\packages\Fody.2.0.6\build\netstandard1.4\Fody.targets" />
+    <Import Project="..\..\packages\Fody.1.29.4\build\dotnet\Fody.targets" Condition="Exists('..\..\packages\Fody.1.29.4\build\dotnet\Fody.targets')" />
```
Comment (fealebenpae) [inline Realm/Realm.Sync/Realm.Sync.csproj]: Indentation.

File: Realm/Realm/Linq/RealmResultsVisitor.cs (hunk 1 above)
Comment (fealebenpae) [inline Realm/Realm/Linq/RealmResultsVisitor.cs]: Could we compare the `MethodHandle` property on `MethodInfo` directly?

Comment (nirinchev) [inline Realm/Realm/Linq/RealmResultsVisitor.cs]: It looks like `MethodHandle` is not available in .NET Standard 1.4 😞

**Pull request merged**